src/Hedgehog.Experimental/Gen.fs

```diff
@@ -44,6 +44,29 @@ module GenX =
     let cString (lower : int) (upper : int) : (Gen<char> -> Gen<string>) =
         Gen.string (Range.constant lower upper)
 
+    /// Generates a permutation of the given list.
+    // "Inside-out" algorithm of Fisher-Yates shuffle from https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle#The_%22inside-out%22_algorithm
+    let shuffle (xs: 'a list) =
+        gen {
+            let shuffled = Array.zeroCreate<'a>(xs.Length)
+            for i = 0 to xs.Length - 1 do
+                let! j = Gen.integral (Range.constant 0 i)
+                if i <> j then shuffled.[i] <- shuffled.[j]
+                shuffled.[j] <- xs.[i]
+            return shuffled |> Array.toList
+        }
```

Comment:
We can also take into account the [`shuffle`](https://github.com/hedgehogqa/haskell-hedgehog/blob/891b67c85d637441e76a0151ac02f1eb63006a05/hedgehog/src/Hedgehog/Internal/Gen.hs#L1502-L1512) the Haskell version of Hedgehog.